### PR TITLE
feat(searchForFacetValues): fall back to client.search if it's present

### PR DESCRIPTION
**Pull request by Haroenv**
client.search with `type: 'facet', facet, params: { facetQuery }` can be used in v4 onwards as an alternative to client.searchForFacetValues, meaning the client.searchForFacetValues method can be removed in a future major version.

Unfortunately the JS client v3 doesn't support type "facet" for multi-index search because it manually decides which keys are used in the post object: https://github.com/algolia/algoliasearch-client-javascript/blob/v3/src/AlgoliaSearchCore.js#L638-L654

FX-1432

### CODE REVIEW

File: src/algoliasearch.helper.js [hunk 1]
```diff
@@ -330,31 +330,58 @@ AlgoliaSearchHelper.prototype.findAnswers = function(options) {
...
+    ]);
+    // algoliasearch < 3.27.1
+  } else if (clientHasInitIndex) {
+    searchForFacetValuesPromise = this.client
+      .initIndex(state.index)
+      .searchForFacetValues(algoliaQuery);
+    // algoliasearch ~5.0.0
+  } else {
+    // @MAJOR only use client.search
+    delete algoliaQuery.facetName;
```
Comment (dhayab) [inline src/algoliasearch.helper.js:364]: Why is this removed from the params payload?

**Comment by Haroenv** [comment]
Marked as a draft as the signature for v5 and v4 aren't equal wrt params

**Review by dhayab — APPROVED** [review]

**Review by francoischalifour — APPROVED** [review]
